File: server/src/battle/model/RandomCustom.java
package battle.model;

import util.battle.BattleUtils;

public class RandomCustom {

    public int MODULE = (int) Math.pow(2,16) + 1; //m
    private int multiplier = 75; //a
    private int increment = 74; //c
    private int seed;

    public RandomCustom(int seedRandom) {
        this.seed = seedRandom;
    }

    public double getRandom() {
        this.seed = (multiplier * this.seed + increment) % MODULE;
        return BattleUtils.round((double) seed / MODULE);
    }
}

File: client/src/modules/base/RandomCustom.js
var RandomCustom = cc.Class.extend({

    MODULE: Math.pow(2,16) + 1, //m
    multiplier: 75, //a
    increment: 74, //c

    ctor: function (seedRandom) {
        this.seed = seedRandom;
    },

    getRandom: function () {
        this.seed = (this.multiplier * this.seed + this.increment) % this.MODULE;
        return Utils.round(this.seed / this.MODULE);
    }
})

